dom/src/main/scala-3/fs2/dom/Dom.scala

```diff
@@ -48,6 +52,190 @@ object Node {
+      new WrappedRef {
+        def unsafeGet() = node.innerText
+        def unsafeSet(s: String) = node.innerText = s
+      }
+  }
+
+  val ENTITY_REFERENCE_NODE: Int = dom.Node.ENTITY_REFERENCE_NODE
+
+  val ATTRIBUTE_NODE: Int = dom.Node.ATTRIBUTE_NODE
+
+  val DOCUMENT_FRAGMENT_NODE: Int = dom.Node.DOCUMENT_FRAGMENT_NODE
+
+  val TEXT_NODE: Int = dom.Node.TEXT_NODE
+
+  val ELEMENT_NODE: Int = dom.Node.ELEMENT_NODE
+
+  val COMMENT_NODE: Int = dom.Node.COMMENT_NODE
+
+  val DOCUMENT_POSITION_DISCONNECTED: Int = dom.Node.DOCUMENT_POSITION_DISCONNECTED
+
+  val DOCUMENT_POSITION_CONTAINED_BY: Int = dom.Node.DOCUMENT_POSITION_CONTAINED_BY
+
+  val DOCUMENT_POSITION_CONTAINS: Int = dom.Node.DOCUMENT_POSITION_CONTAINS
+
+  val DOCUMENT_TYPE_NODE: Int = dom.Node.DOCUMENT_TYPE_NODE
+
+  val DOCUMENT_POSITION_IMPLEMENTATION_SPECIFIC: Int =
+    dom.Node.DOCUMENT_POSITION_IMPLEMENTATION_SPECIFIC
+
+  val DOCUMENT_NODE: Int = dom.Node.DOCUMENT_NODE
+
+  val ENTITY_NODE: Int = dom.Node.ENTITY_NODE
+
+  val PROCESSING_INSTRUCTION_NODE: Int = dom.Node.PROCESSING_INSTRUCTION_NODE
+
+  val CDATA_SECTION_NODE: Int = dom.Node.CDATA_SECTION_NODE
+
+  val NOTATION_NODE: Int = dom.Node.NOTATION_NODE
+
+  val DOCUMENT_POSITION_FOLLOWING: Int = dom.Node.DOCUMENT_POSITION_FOLLOWING
+
+  val DOCUMENT_POSITION_PRECEDING: Int = dom.Node.DOCUMENT_POSITION_PRECEDING
+}
+
+opaque type DOMList[F[_], +A] = dom.DOMList[A]
+
+object DOMList {
+  extension [F[_], A <: Node[F]](domList: DOMList[F, A]) {
+    def apply(index: Int)(using F: Dom[F]): F[A] = F.delay(domList(index))
+    def length(using F: Dom[F]): F[Int] = F.delay(domList.length)
+    def toList: List[A] = dom.DOMList.domListAsSeq(domList).toList
+  }
+}
+
+opaque type NodeList[F[_], +A <: Node[F]] <: DOMList[F, A] = dom.NodeList[A]
+
+object NodeList {
+  extension [F[_], A <: Node[F]](nodeList: NodeList[F, A]) {
+    def item(index: Int)(using F: Dom[F]): F[A] = F.delay(nodeList.item(index))
+  }
+}
+
+opaque type NamedNodeMap[F[_]] = dom.NamedNodeMap
+
+object NamedNodeMap {
+  extension [F[_]](nodeMap: NamedNodeMap[F]) {
+    def length(using F: Dom[F]): F[Int] = F.delay(nodeMap.length)
+
+    def removeNamedItemNS(namespaceURI: String, localName: String)(using F: Dom[F]): F[Attr[F]] =
+      F.delay(nodeMap.removeNamedItemNS(namespaceURI, localName))
+
+    def item(index: Int)(using F: Dom[F]): F[Attr[F]] = F.delay(nodeMap.item(index))
+
+    def apply(index: Int)(using F: Dom[F]): F[Attr[F]] = F.delay(nodeMap(index))
+
+    def update(index: Int, v: Attr[F])(using F: Dom[F]): F[Unit] = F.delay(nodeMap.update(index, v))
+
+    def removeNamedItem(name: String)(using F: Dom[F]): F[Attr[F]] =
+      F.delay(nodeMap.removeNamedItem(name))
+
+    def getNamedItem(name: String)(using F: Dom[F]): F[Option[Attr[F]]] =
+      F.delay(Option(nodeMap.getNamedItem(name)))
+
+    def setNamedItem(arg: Attr[F])(using F: Dom[F]): F[Attr[F]] = F.delay(nodeMap.setNamedItem(arg))
```

Comment:
get/set here maps directly to that structure, therefore I kept it for now. Also, get is partial.